perception/ground_segmentation/test/test_scan_ground_filter.cpp

```diff
@@ -0,0 +1,170 @@
+
+    // input cloud frame MUST be base_link
+    input_msg_ptr_ = std::make_shared<sensor_msgs::msg::PointCloud2>();
+    geometry_msgs::msg::TransformStamped t;
+    t.header.frame_id = "base_link";
+    t.child_frame_id = "velodyne_top";
+    t.transform.translation.x = 0.6;
+    t.transform.translation.y = 0;
+    t.transform.translation.z = 2;
+    tf2::Quaternion q;
+    q.setRPY(0.0, 0.0, 0.0);
+    t.transform.rotation.x = q.x();
+    t.transform.rotation.y = q.y();
+    t.transform.rotation.z = q.z();
+    t.transform.rotation.w = q.w();
+
+    tf2::doTransform(*origin_input_msg_ptr, *input_msg_ptr_, t);
+
+    parse_yaml();
+  }
+
+  ScanGroundFilterTest() {}
+
+  ~ScanGroundFilterTest() override { rclcpp::shutdown(); }
+
+public:
+  std::shared_ptr<ground_segmentation::ScanGroundFilterComponent> scan_ground_filter_;
+  rclcpp::Node::SharedPtr dummy_node_;
+  rclcpp::Publisher<sensor_msgs::msg::PointCloud2>::SharedPtr input_pointcloud_pub_;
+  rclcpp::Publisher<sensor_msgs::msg::PointCloud2>::SharedPtr output_pointcloud_pub_;
+
+  sensor_msgs::msg::PointCloud2::SharedPtr input_msg_ptr_;
+
+  // wrapper function to test private function filter
+  void filter(sensor_msgs::msg::PointCloud2 & out_cloud)
+  {
+    scan_ground_filter_->filter(input_msg_ptr_, nullptr, out_cloud);
+  }
+
+  void parse_yaml()
+  {
+    const auto share_dir = ament_index_cpp::get_package_share_directory("tier4_perception_launch");
+    const auto config_path =
+      share_dir +
+      "/config/obstacle_segmentation/ground_segmentation/ground_segmentation.param.yaml";
+    std::cout << "config_path:" << config_path << std::endl;
+    YAML::Node config = YAML::LoadFile(config_path);
+    auto params = config["/**"]["ros__parameters"]["common_ground_filter"]["parameters"];
+    global_slope_max_angle_deg_ = params["global_slope_max_angle_deg"].as<float>();
+    local_slope_max_angle_deg_ = params["local_slope_max_angle_deg"].as<float>();
+    split_points_distance_tolerance_ = params["split_points_distance_tolerance"].as<float>();
+    split_height_distance_ = params["split_height_distance"].as<float>();
+  }
+
+  float global_slope_max_angle_deg_ = 0.0;
+  float local_slope_max_angle_deg_ = 0.0;
+  float split_points_distance_tolerance_ = 0.0;
+  float split_height_distance_ = 0.0;
+};
+
+TEST_F(ScanGroundFilterTest, TestCase1)
+{
+  input_pointcloud_pub_->publish(*input_msg_ptr_);
+  sensor_msgs::msg::PointCloud2 out_cloud;
+
+  // set filter parameter
+  scan_ground_filter_->set_parameter(
+    rclcpp::Parameter("global_slope_max_angle_deg", global_slope_max_angle_deg_));
+  scan_ground_filter_->set_parameter(
+    rclcpp::Parameter("local_slope_max_angle_deg", local_slope_max_angle_deg_));
+  scan_ground_filter_->set_parameter(
+    rclcpp::Parameter("split_points_distance_tolerance", split_points_distance_tolerance_));
+  scan_ground_filter_->set_parameter(
+    rclcpp::Parameter("split_height_distance", split_height_distance_));
+
```

Comment:
Thank you for your update, Since we have added some new parameters for scan_ground_filter [here](https://github.com/autowarefoundation/autoware.universe/pull/1899), could you add new parameters with this PR.